src/lib/es2015.iterable.d.ts

```diff
@@ -8,23 +8,30 @@ interface SymbolConstructor {
     readonly iterator: symbol;
 }
 
-interface IteratorResult<T> {
-    done: boolean;
-    value: T;
+interface IteratorYieldResult<TYield> {
+    done: false;
+    value: TYield;
 }
 
-interface Iterator<T> {
-    next(value?: any): IteratorResult<T>;
-    return?(value?: any): IteratorResult<T>;
-    throw?(e?: any): IteratorResult<T>;
+interface IteratorReturnResult<TReturn> {

```

Comment:
`value` could be omitted in this case

```
interface IteratorReturnResult {
    done: true;
}
```

Reply:
Unfortunately, for accurate type information for `yield*` it cannot be made optional. While it may be true that when you *produce* an `IteratorResult` you might want to elide `value`, when you *consume* an `IteratorResult`, the type system interrogates the `value` property to determine the return type. If the `value` property was optional, then we would always infer `TReturn | undefined` (due to the optionality). Since we cannot specialize `IteratorReturnResult` for both cases, we must leave `value` as mandatory to infer the correct type.

Reply:
That's ok. It's not so important.